sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerIO.java

```diff
@@ -123,15 +123,16 @@ public static Write write() {
   public abstract static class Write extends PTransform<PCollection<Mutation>, PDone> {
 
     @Nullable
-    abstract String getProjectId();
+    abstract ValueProvider<String> getProjectId();
 
     @Nullable
-    abstract String getInstanceId();
+    abstract ValueProvider<String> getInstanceId();
 
     @Nullable
-    abstract String getDatabaseId();
+    abstract ValueProvider<String> getDatabaseId();
 
-    abstract long getBatchSizeBytes();
+    @Nullable
+    abstract ValueProvider<Long> getBatchSizeBytes();
```

Comment:
Got it. I don't think `batch size` would be something that we let people configure in templates. Removed the `ValueProvider`.